core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -0,0 +1,77 @@
+package org.infinispan;
+
+import java.util.Collection;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+
+import org.infinispan.configuration.cache.TransactionConfiguration;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.util.function.SerializableBiConsumer;
+import org.infinispan.util.function.SerializablePredicate;
+
+/**
+ * Stream that allows for operation upon data solely with side effects by using {@link LockedStream#forEach(BiConsumer)}
+ * where the <b>BiConsumer</b> is invoked while guaranteeing that the entry being passed is properly locked for the
+ * entire duration of the invocation.
```

Comment:
Not sure if I follow. I've intended to point out that it's a bit ambiguous if you should use `cache.put(...)` or `cacheEntry.setValue(...)` for the modification (or both); in the tests you use the former, so I assume that this is it. This pattern should be mentioned in the javadoc to make it clear for those that don't read testsuite :)
In addition to that, `cacheEntry.setValue(...)` et all should either transparently call `cache.put(...)` (but that might be unsuitable as the user could want to set metadata as well), or throw an exception.